lib/shoryuken/manager.rb

```diff
@@ -57,8 +57,12 @@ def ready
       @max_processors - busy
     end
 
-    def processor_done
+    def processor_done(queue)
       @busy_processors.decrement
+      client_queue = Shoryuken::Client.queues(queue)
+      return unless client_queue.fifo?
+
+      @polling_strategy.unpause_queue(queue)
```

Comment:
Since [users can specify their own polling strategy class][0], we want to make it optional to implement the new method. Let's invoke this method only if `@polling_strategy` responds to it.

[0]: https://github.com/cjlarose/shoryuken/blob/1a261d66a678a41266e389642c54bbf07819ee21/lib/shoryuken/options.rb#L65-L66